### PR TITLE
Sync workspace member across devices after removing member(s)

### PR DESCRIPTION
**Pull request by K4tsuki**
### Details

Sync workspace member across device after member(s) removed.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7095

### Tests

- [x] Verify that no errors appear in the JS console

1. Open workspace member page with same account on two device
2. Remove member(s) on a device and refresh member page in other device
3. Members of workspace must be same on those two device
4. In android must go back to setting page and then go to member page to update the member list.

### QA Steps

- [x] Verify that no errors appear in the JS console

1. Open workspace member page with same account on two device
2. Remove member(s) on a device and refresh member page in other device
3. Members of workspace must be same on those two device
4. In android must go back to setting page and then go to member page to update the member list.

### Tested On

- [x] Web
- [x] Mobile Web
- [ ] Desktop
- [ ] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/1313124/151729076-e7d437b3-51b1-4dc9-87f5-e146f4b476f2.mp4

#### Mobile Web

https://user-images.githubusercontent.com/1313124/151729110-04065df4-ff0b-499c-85f6-80e8a625601a.mp4

#### Android

https://user-images.githubusercontent.com/1313124/151729126-27f8df6a-67f1-4e2a-a7dc-6f5cec888f0b.mp4

### CODE REVIEW

File: src/libs/actions/Policy.js [hunk 1]
```diff
@@ -223,7 +223,13 @@ function loadFullPolicy(policyID) {
...
             }
 
-            Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`, getSimplifiedPolicyObject(policy));
+            const key = `${ONYXKEYS.COLLECTION.POLICY}${policyID}`;
+            const currentPolicy = allPolicies[key];
+
+            const simplifiedPolicyObject = getSimplifiedPolicyObject(policy);
+            const updatedPolicyObject = {...currentPolicy, ...simplifiedPolicyObject};
+
+            Onyx.set(`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`, updatedPolicyObject);
```
Comment (parasharrajat) [inline src/libs/actions/Policy.js:232]:
```suggestion
             Onyx.set(`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`, {
            	... allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`],
            	...getSimplifiedPolicyObject(policy),
            });
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by K4tsuki** [comment]
Fix linter

**Review by parasharrajat — DISMISSED** [review]
Works fine for the linked issue. 

There could be unintended side-effects of this change. `withFullPolicy` is used on all workspace pages. we use `Onyx.merge` to merge the changes but for this we are using `Onyx.set`. There are `limitations` of using `Onyx.set`. It cancels the `merge` requests which could interfere with other Workspace pages operations. **This is an assumption so It is possible that it never happens**. 

But Onyx poses a challenge for us and it does not give us any other choice to fix the linked issue ATM. So I am approving this. 

cc: @Luke9389 

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by K4tsuki** [comment]
cc: @Luke9389  All checks have passed

**Comment by Luke9389** [comment]
> There could be unintended side-effects of this change.

@parasharrajat, this concerns me. Before approving I'm going to go read up on Onyx and be sure this change is exactly what we want it to be.

**Comment by K4tsuki** [comment]
@Luke9389  We have been using Onyx.set in `removeMember` and there was no problem because of this.

https://github.com/Expensify/App/blob/16d38de1a943bc8393d34b745898f9d546a9e5be/src/libs/actions/Policy.js#L299

Also Kidroca tell to us to using Onyx.set in this issue in slack conversation. 

I have tested it on navigating between workspaces, add and remove members and there is no problem.

**Review by Luke9389 — CHANGES_REQUESTED** [review]
Looks like there's now a merge conflict. Please merge main

**Comment by K4tsuki** [comment]
@Luke9389 main merged.

**Review by Luke9389 — APPROVED** [review]
After reviewing some materials, I think this is fine.

**Pull request merged**